src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -376,9 +377,9 @@ void BaseCouplingScheme::secondExchange()
 
     // Update internal time tracking
     if (_isTimeWindowComplete) {
-      _timeWindowStartTime += _timeWindowSize;
+      _timeWindowStartTime(_timeWindowSize);
     }
-    _time           = _timeWindowStartTime;
+    _time           = getWindowStartTime();
```

Comment:
For consistency reasons, I would consider to make the (intermediate buffer) `_time` also such an accumulator, as we store the subcycling substeps in it.

Reply:
I think we will only be able to solve the problem that triggered this whole discussion, if we turn `_time` into such a buffer. `_timeWindowStartTime` is a nice-to-have (probably similar problem on a different level), but `_time` is the actual problem. But let's first add a test and then we will know.